File: binarysearch/Index-into-an-Infinite-String/Index-into-an-Infinite-String.cpp
// https://helloacm.com/algorithms-to-compute-the-substring-with-indexing-into-an-infinite-string/
// https://binarysearch.com/problems/Index-into-an-Infinite-String
// EASY, STRING

string solve(string s, int i, int j) {
    int n = s.size();
    string ans = "";
    for (int k = i; k < j; ++ k) {
        ans += s[k % n];
    }
    return ans;
}

File: binarysearch/Index Into an Infinite String/Index Into an Infinite String.py
# https://helloacm.com/teaching-kids-programming-index-into-an-infinite-string-find-substring/
# https://binarysearch.com/problems/Index-Into-an-Infinite-String
# EASY, MATH

class Solution:
    def solve(self, s, i, j):
        return "".join(s[x % len(s)] for x in range(i, j))
